src/core/forms.py

```diff
@@ -26,9 +25,12 @@ class Meta:
 
 
 class TransactionForm(forms.ModelForm):
+
     def __init__(self, *args, **kwargs):
+        self.user = kwargs.pop('user')
         super().__init__(*args, **kwargs)
         self.fields['category'].empty_label = "Category not selected"
+        self.fields['category'].queryset = Category.objects.filter(user=self.user)
```

Comment:
I think that a better way is just, directly set kwargs.pop['user'], instead setting user as object variable.